components/core/src/clp/ffi/ir_stream/protocol_constants.hpp

```diff
@@ -67,11 +67,13 @@ constexpr int8_t ValueEightByteEncodingClpStr = 0x5A;
 constexpr int8_t ValueEmpty = 0x5E;
 constexpr int8_t ValueNull = 0x5F;
 
-constexpr int8_t SchemaTreeNodeParentIdUByte = 0x60;
-constexpr int8_t SchemaTreeNodeParentIdUShort = 0x61;
+constexpr int8_t EncodedSchemaTreeNodeParentIdByte = 0x60;
+constexpr int8_t EncodedSchemaTreeNodeParentIdShort = 0x61;
+constexpr int8_t EncodedSchemaTreeNodeParentIdInt = 0x62;
 
-constexpr int8_t KeyIdUByte = 0x65;
-constexpr int8_t KeyIdUShort = 0x66;
+constexpr int8_t EncodedKeyIdByte = 0x65;
+constexpr int8_t EncodedKeyIdShort = 0x66;
+constexpr int8_t EncodedKeyIdInt = 0x67;
```

Comment:
Sorry, I didn't notice this in the doc but I did notice it now---why are we adding support for 4-byte node IDs?

Reply:
Since we change the way of how node IDs are encoded, the range of node IDs supported also changed (dropped from max=65535 to max=32767). The 4-byte node ID tags are already reserved in the protocol, we just didn't implement them in our first iteration. Adding it with this PR in case 32767 isn't enough to cover all schema tree nodes. I thought Devin had same concern before and he mentioned maybe we should add support for 4-byte ID encodings. However, with the auto-generated IDs supported, arguably we could shift some kv pairs to the auto-generated namespace, so maybe 32767 should be a reasonably large space...